server/mcp_server_mobile_use/internal/mobile_use/server/server.go

```diff
@@ -0,0 +1,219 @@
+	mcpServer.AddTool(appUploadTool, tool.HandleAppUploadTool())
+
+	appLaunchTool := tool.NewLaunchAppTool()
+	mcpServer.AddTool(appLaunchTool, tool.HandleLaunchAppTool())
+
+	appCloseTool := tool.NewCloseAppTool()
+	mcpServer.AddTool(appCloseTool, tool.HandleCloseAppTool())
+
+	listAppsTool := tool.NewListAppTool()
+	mcpServer.AddTool(listAppsTool, tool.HandleListAppTool())
+
+	keyEventBackTool := tool.NewKeyEventBackTool()
+	mcpServer.AddTool(keyEventBackTool, tool.HandleKeyEventBackTool())
+
+	keyEventHomeTool := tool.NewKeyEventHomeTool()
+	mcpServer.AddTool(keyEventHomeTool, tool.HandleKeyEventHomeTool())
+
+	keyEventMenuTool := tool.NewKeyEventMenuTool()
+	mcpServer.AddTool(keyEventMenuTool, tool.HandleKeyEventMenuTool())
+
+	ctx, cancel := context.WithCancel(context.Background())
+	return &MobileUseServer{
+		server:      mcpServer,
+		shutdownCtx: ctx,
+		cancel:      cancel,
+		doneCh:      make(chan struct{}),
+	}
+}
+
+func (s *MobileUseServer) StartSSE(baseUrl string) *mcp_srv.SSEServer {
+	return mcp_srv.NewSSEServer(s.server,
+		mcp_srv.WithBaseURL(baseUrl),
+		mcp_srv.WithSSEContextFunc(authFromRequest),
+	)
+}
+
+// StartStdio starts the MCP server in stdio mode
+// It returns a done channel that is closed when the stdio server exits
+func (s *MobileUseServer) StartStdio() error {
+	go func() {
+		err := mcp_srv.ServeStdio(s.server, mcp_srv.WithStdioContextFunc(authFromEnv))
+		s.cancel()      // Cancel context when stdio exits
+		close(s.doneCh) // Signal that stdio is done
+		if err != nil {
+			// Log error or handle it somehow
+		}
+	}()
+	return nil
+}
+
+// WaitForDone blocks until the server is done (either shutdown by signal or completed naturally)
+func (s *MobileUseServer) WaitForDone() error {
+	select {
+	case <-s.doneCh:
+		return nil
+	case <-s.shutdownCtx.Done():
+		return errors.New("server shutdown")
+	}
+}
+
+// StartSSEWithServer starts the SSE server with a graceful shutdown capability
+func (s *MobileUseServer) StartSSEWithServer(addr string, baseUrl string) error {
+	sseServer := s.StartSSE(baseUrl)
+
+	// Start HTTP server in a goroutine
+	go func() {
+		err := sseServer.Start(addr)
+		if err != nil && !errors.Is(err, http.ErrServerClosed) {
+			// If error is not due to server shutdown, signal server is done
+			s.cancel()
+		}
+		close(s.doneCh)
+	}()
+
+	return nil
+}
+
+// Shutdown gracefully stops the server
+func (s *MobileUseServer) Shutdown() {
+	s.cancel()
+	s.shutdownWg.Wait()
+}
+
+func authFromEnv(ctx context.Context) context.Context {
+	accessKeyId := os.Getenv("ACEP_ACCESS_KEY")
+	accessKeySecret := os.Getenv("ACEP_SECRET_KEY")
+	productId := os.Getenv("ACEP_PRODUCT_ID")
+	deviceId := os.Getenv("ACEP_DEVICE_ID")
+	acepHost := os.Getenv("ACEP_HOST")
+	tosAccessKey := os.Getenv("TOS_ACCESS_KEY")
+	tosSecretKey := os.Getenv("TOS_SECRET_KEY")
+	bucket := os.Getenv("ACEP_TOS_BUCKET")
+	region := os.Getenv("ACEP_TOS_REGION")
+	endpoint := os.Getenv("ACEP_TOS_ENDPOINT")
```

Comment:
Is ACEP_ACCESS_KEY equal VOLCENGINE_ACCESS_KEY? validate the naming convention of envs variable.